### ISSUE TITLE
FR: Show `@` in listed commits after `jj commit`

### ISSUE THREAD

**Issue by robinst** [jj-vcs/jj#4700]
**Is your feature request related to a problem? Please describe.**

After doing a `jj commit`, the output shows 2 commits, like this:

```
$ jj commit -i
Working copy now at: mnnxpsqr 4a8c5c20 (no description set)
Parent commit      : szvnmmku 19143488 Test commit
```

I'm still very new to jj, and what I sometimes want to do after the commit is to set a bookmark, so I do:

```
jj bookmark set mybookmark -r @-
```

However, I've found myself to be not 100% sure that `@-` refers to the right commit there.

**Describe the solution you'd like**

I think it would help if the output indicated which one is `@` (and maybe even which one is `@-`), e.g. like this:

```
$ jj commit -i
Working copy now at: @  mnnxpsqr 4a8c5c20 (no description set)
Parent commit      : @- szvnmmku 19143488 Test commit
```

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

**Comment by martinvonz** [comment]
That's unusual formatting. I wouldn't mind something like this instead:
```
Working copy (@) now at: mnnxpsqr 4a8c5c20 (no description set)
Parent commit (@-)     : szvnmmku 19143488 Test commit
```

**Comment by ilyagr** [comment]
Interesting idea.

One way to make it a bit shorter (same length as we have it currently):

```
Working copy now @ : mnnxpsqr 4a8c5c20 (no description set)
Parent commit    @-: szvnmmku 19143488 Test commit
```

It's slightly cutesy with the `@=at`, but maybe that's OK?

Martin's example looks a bit ugly to me because of the misalignment, but I like the spirit of it as well. One possibility is just to add a space:

```
Working copy  (@) now at: mnnxpsqr 4a8c5c20 (no description set)
Parent commit (@-)      : szvnmmku 19143488 Test commit
```

Another possibility that comes to mind:


```
Working copy now at: mnnxpsqr (@)  4a8c5c20 (no description set)
Parent commit      : szvnmmku (@-) 19143488 Test commit
```

with parentheses in white and `@`/`@-` in green.

-----

At the moment, all of these look a bit off to me, but it's very possible I'd get used to any of them quickly.

**Comment by ilyagr** [comment]
If we change the output after mutating commands, we'd probably want to also adjust `jj status` which currently looks like

```
Working copy : mnnxpsqr 4a8c5c20 (no description set)
Parent commit: szvnmmku 19143488 Test commit
```

**Comment by ilyagr** [comment]
I also wonder whether @robinst 's original suggestion would look fine with color (`@` and `@-` green).

**Comment by martinvonz** [comment]
> It's slightly cutesy with the `@=at`, but maybe that's OK?

I would not have picked up on that if you had not pointed it out but that "at" is a bit inconsistent as it is anyway (there's no "at" after "Parent commit").

> Another possibility that comes to mind:
> 
> ```
> Working copy now at: mnnxpsqr (@)  4a8c5c20 (no description set)
> Parent commit      : szvnmmku (@-) 19143488 Test commit
> ```

That would be more complicated to implement and configure because we'd need a new template for "commit summary with @/@-".

**Comment by ilyagr** [comment]
> That would be more complicated to implement and configure because we'd need a new template for "commit summary with @/@-".

It might still be worth it, but that's a good point. I now think that `@` and `@-` would go well together with the list if branches.

**Comment by ilyagr** [comment]
BTW, I'm not sure how this would work if there are multiple parent commits. Maybe we should only add `@` to the working copy, and not have `@-`?

**Comment by martinvonz** [comment]
> BTW, I'm not sure how this would work if there are multiple parent commits. Maybe we should only add `@` to the working copy, and not have `@-`?

We display multiple rows. I was thinking it would be fine to repeat the `@-` for all of them.

**Comment by ilyagr** [comment]
> I was thinking it would be fine to repeat the @- for all of them.

I'm not sure, but perhaps you are right. 

--------

Re implementation difficulty, where would we use "commit summary without @"?

For a much more complex solution, we could in principle add an analogue of CSS's `display: none` to the formatter. So, we'd have a label `at` and formatter rules 

```toml
"commit_summary at" = {hide = true}
"wc_and_parent commit_summary at" = {hide = false}
```

**Comment by martinvonz** [comment]
> Re implementation difficulty, where would we use "commit summary without @"?

We currently use that in many places (I think you can search the code for `commit_summary`). Including `@` and `@-` would be redundant at least in the `jj log` output.

**Comment by martinvonz** [comment]
I like the form where we put `@` and `@-` before the colon anyway, and it's simpler, so that's definitely how I would implement it.

**Comment by robinst** [comment]
> That's unusual formatting.

Note that my proposal put it before the change ID to mimic `jj log` output, which looks like this:

```
@  mnnxpsqr robinst@... 2024-10-24 16:49:04 4a50f1d6
│  (no description set)
◆  szvnmmku robinst@... 2024-10-24 11:39:11 ...
```

But yeah, the analogy already breaks down with the parent commit, so not sure.

+1 to using the same color for `@` as `jj log`.

**Comment by martinvonz** [comment]
> Note that my proposal put it before the change ID to mimic `jj log` output, which looks like this:

Yes, I understood that. Sorry about the short "That's unusual formatting." without further explanation. What I meant was that the stuff after the colon is exactly the `commit_summary` template. I would prefer to continue to use just that there.

**Comment by robinst** [comment]
Looked into this now and raised a PR for it here:
- https://github.com/jj-vcs/jj/pull/5891
